lib/registries/addon/components/editable-field/tags-manager/component.ts

```diff
@@ -0,0 +1,93 @@
+import { tagName } from '@ember-decorators/component';
+import { action, computed } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import Component from '@ember/component';
+import config from 'ember-get-config';
+
+import { layout } from 'ember-osf-web/decorators/component';
+import Registration from 'ember-osf-web/models/registration';
+import Analytics from 'ember-osf-web/services/analytics';
+import pathJoin from 'ember-osf-web/utils/path-join';
+import template from './template';
+
+const { OSF: { url: baseUrl } } = config;
+
+@tagName('')
+@layout(template)
+export default class TagsManager extends Component {
+    // required
+    registration!: Registration;
+
+    // private properties
+    @service analytics!: Analytics;
+
+    currentTags: string[] = [];
+
+    constructor(properties: object) {
+        super(properties);
+        if (this.registration) {
+            this.setProperties({ currentTags: [...this.registration.tags] });
+        }
+    }
+
+    @computed('registration.tags.length')
+    get fieldIsEmpty() {
+        return !this.registration.tags.length;
+    }
+
+    @computed('currentTags.length,registration.{tags.length,hasDirtyAttributes}')
```

Comment:
you are right `registration.tags.[]` is what should be used here. this computed won't be needed since we will always show `save` and `cancel` buttons.